### ISSUE TITLE
Unparsing error on a date function

### ISSUE THREAD

**Issue by Shrpp** [sqlfluff/sqlfluff#3958]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I've tried to reproduce `lint`, but it give me an unparsing error.

### Expected Behaviour

I was expecting the tool to tell me about possible errors in the `.sql`.

### Observed Behaviour

The tool gave me an unparsing error concerning a basic date function from Big Query

`extract(week(monday) from date)`

Log:

```
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [models\marts\yaigo\weekly\weekly_partner_collection.sql] FAIL
L:  21 | P:  14 |  PRS | Line 17, Position 14: Found unparsable section:
                       | '(week(monday) from orders.fecha_entrega)...'
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?
```

### How to reproduce

Running `sqlfluff lint`, because the fix statement doesn't recognize the error on the .sql

### Dialect
```
dialect = bigquery
templater = dbt
```
### Version
```
python 3.9.12
sqlfluff == 1.3.0
sqlfluff-templater-dbt == 1.3.0
dbt-core == 1.3.0
dbt-bigquery == 1.3.0
```

### Configuration
```
[sqlfluff]
templater = dbt
dialect = bigquery
rules = L001, L002, L006, L011, L023, L024, L035, L039, L041
encoding = utf-8

[sqlfluff:rules]
capitalisation_policy = lower
tab_space_size = 2
indent_unit = space

[sqlfluff:indentation]
indented_joins = true
indented_using_on = true

[sqlfluff:templater:jinja]
apply_dbt_builtins = true

[sqlfluff:templater:jinja:macros]
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}

[sqlfluff:templater:dbt]
project_dir = .
profile_dir = ./.
```
### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by adam-tokarski** [comment]
Hi @Shrpp, please provide example file for which you've observed the issue. 
I've checked with `EXTRACT(WEEK...` example from bigquery docs and it just works.

**Comment by Shrpp** [comment]
Hey @adam-tokarski! Sure here's the example file.

```sql
    , extract( week( monday) from orders.fecha_entrega) as semana
    , cast( date_trunc( cast(orders.fecha_entrega as date),week( monday)) as string) as fecha_desde
    , cast( date_add( date_trunc( date_sub( current_date, interval 1 week),week( monday)), interval 6 day)as string) as fecha_hasta

```

**Comment by adam-tokarski** [comment]
@Shrpp is this for real whole example file? Seems like missing something at the begging, like

```sql
 select 
    extract( week( monday) from orders.fecha_entrega) as semana
    , cast( date_trunc( cast(orders.fecha_entrega as date),week( monday)) as string) as fecha_desde
    , cast( date_add( date_trunc( date_sub( current_date, interval 1 week),week( monday)), interval 6 day)as string) as fecha_hasta
```

**Comment by Shrpp** [comment]
Those lines are the problem, the lines beside are ids

**Comment by Shrpp** [comment]
However, I don't think the fix is necessary. I just updated all SQLFluff (both in the CLI and in the Git Hub Action) and it works perfectly now
